woocommerce/woocommerce-gutenberg-products-block
# Refactor: Use an alternative mini cart icon to fix the Mini Cart styling issues

Fixes #5930

The current Mini Cart icon leads to some difficulties and style issues:
- We need the background for the counter (badge) => an extra setting which isn't really necessary.
- The theme block doesn't work great with dark themes (at least, out of the box).
  - For a special case - Twenty Twenty Two - its large header page template uses a dark header, while other pages use the normal white header. The Mini Cart doesn't work with it.
  - Increase code complexity: because we need to style an inner element of the Mini Cart block, we need `__experimentalSkipSerialization` to target inner elements for global style. We also need to calculate manually color classes and style inside the block component.

This PR tries to solve all of the above issues by using the icon as proposed [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5930#issuecomment-1054942322). By using that icon, we can:
- Remove the background setting which is useless now.
- Remove the transparent button option, which doesn't make sense anymore without the background setting.
- Remove `__experimentalSkipSerialization`, let Blocks Editor and Site Editor target the Mini Cart block using `.wp-block-woocommerce-mini-cart`.
- Remove color and class manipulation for the counter.
- Icon inherits the color from the parent by default, so it works with both dark and light backgrounds, and at the same time.




#### Accessibility




#### Other Checks


### Screenshots

<img width="1143" alt="image" src="https://user-images.githubusercontent.com/5423135/156558499-eaf831cc-a607-4cd0-9500-4da85b095704.png">



### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

1. Add the block to the header.
2. Make sure it works by default with Twenty Twenty Two. On both dark and light headers.
3. Test the predefined color and custom color for block style and global style.
4. Ensure the block style priority is higher than global style.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

## Notes

- Look like the Site Editor is having an issue clearing the global style cache. Changing the global text color for the Mini Cart icon works after load. But it doesn't work if we keep the editor open:
  - Go to Site Editor > Style (Beta) > Blocks Styles > Mini Cart.
  - Change the text color > Save it.
  - Keep the current tab open, in a new tab, go to the site front-end, see the text color has been updated correctly.
  - Switch to the site editor tab, change the text color of the mini cart to another, see it work on the Editor > Save.
  - Switch to the front end tab, reload the editor, see the color still the previous value.
